Enable RTC in Dlab Hub

balajialg opened the pull request:
As a next step from today's meeting, closed the old PR and created a new PR with all the required changes from the dlab hub.

balajialg committed:
Making changes to Dlab hub to enable RTC

balajialg committed:
Change image name in env.yml file

shaneknapp commented on deployments/dlab/hubploy.yaml:
```diff
@@ -1,7 +1,6 @@
   images:
-    - name: us-central1-docker.pkg.dev/ucb-datahub-2018/user-images/primary-user-image
-      path: ../datahub/images/default
+    - name: us-central1-docker.pkg.dev/ucb-datahub-2018/user-images/dlab-user-image
```
this should look like the following...

```
images:
  image_name: us-central1-docker.pkg.dev/ucb-datahub-2018/user-images/dlab-user-image
```

shaneknapp replied:
just make sure the top of `hubploy.yaml` looks like this:
```
(base) ➜  datahub git:(dlabrtc_branch) ✗ head deployments/dlab/hubploy.yaml
images:
  image_name: us-central1-docker.pkg.dev/ucb-datahub-2018/user-images/dlab-user-image
  registry:
    provider: gcloud
    gcloud:
      project: ucb-datahub-2018
      service_key: gcr-key.json
```

shaneknapp replied:
@balajialg

shaneknapp commented on deployments/dlab/hubploy.yaml:
```diff
@@ -1,7 +1,6 @@
 images:
   images:
```
remove this `images:` line

ryanlovett commented on deployments/dlab/image/environment.yml:
```diff
@@ -0,0 +1,114 @@
+name: dlab-sp23
```
I'd just call this `dlab` since git will do versioning for you, and since the image will probably be perpetually updated. But appending `-sp23` won't do any harm either.

I'd also do a diff between this file and `deployments/stat159/image/environment.yml` and pull of their updates before this merges.

ryanlovett commented on deployments/dlab/config/common.yaml:
```diff
@@ -47,4 +58,4 @@ jupyterhub:
       guarantee: 4G
       limit: 8G
-    image: {}
+    image: {}
```
I think `image: {}` can be removed.

shaneknapp commented:
i tried to hubploy to dlab-staging after building the image w/hubploy (so meta) and got the following error:
```
hubploy deploy dlab hub staging
Activated service account credentials for: [hubploy@ucb-datahub-2018.iam.gserviceaccount.com]
Fetching cluster endpoint and auth data.
kubeconfig entry generated for fall-2019.
Getting updates for unmanaged Helm repositories...
...Successfully got an update from the "https://jupyterhub.github.io/helm-chart/" chart repository
Saving 1 charts
Downloading jupyterhub from repo https://jupyterhub.github.io/helm-chart/
Deleting outdated charts
Error: UPGRADE FAILED: values don't meet the specifications of the schema(s) in the following chart(s):
jupyterhub:
- singleuser.extraFiles.lab-config: mountPath is required

Traceback (most recent call last):
  File "/Users/sknapp/miniconda3/envs/dh/bin/hubploy", line 8, in <module>
    sys.exit(main())
             ^^^^^^
  File "/Users/sknapp/miniconda3/envs/dh/lib/python3.11/site-packages/hubploy/__main__.py", line 128, in main
    helm.deploy(
  File "/Users/sknapp/miniconda3/envs/dh/lib/python3.11/site-packages/hubploy/helm.py", line 171, in deploy
    helm_upgrade(
  File "/Users/sknapp/miniconda3/envs/dh/lib/python3.11/site-packages/hubploy/helm.py", line 95, in helm_upgrade
    subprocess.check_call(cmd)
  File "/Users/sknapp/miniconda3/envs/dh/lib/python3.11/subprocess.py", line 413, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['helm', 'upgrade', '--wait', '--install', '--namespace', 'dlab-staging', 'dlab-staging', 'hub', '-f', 'deployments/dlab/config/common.yaml', '-f', 'deployments/dlab/config/staging.yaml', '-f', '/var/folders/7l/bjg7qcmx6h98xhmxxpy5gcq80000gn/T/tmpfut0darq', '--set-string', 'jupyterhub.singleuser.image.tag=a1f37bf', '--set-string', 'jupyterhub.singleuser.image.name=us-central1-docker.pkg.dev/ucb-datahub-2018/user-images/dlab-user-image']' returned non-zero exit status 1.
```

ryanlovett commented:
@shaneknapp The problem is https://github.com/berkeley-dsep-infra/datahub/blob/a1f37bfe481c6180e2e1e753a9c8e50f2f04a267/deployments/dlab/config/common.yaml#L47. I'd remove the extraFiles section.

balajialg committed:
Remove configs related to Jupyter Link Share

balajialg commented:
@ryanlovett Removed the configs related to Jupyter Link Share. Is it not mandatory config for enabling RTC?

ryanlovett commented:
@balajialg Maybe it was part of an older config? It is not in current stat159 config/common.

balajialg commented:
@ryanlovett that's helpful. Saw through git diff that there isn't any change between the common.yml file and env.yml file between both the hubs. Also, changed the name for the env.yml file

shaneknapp commented:
@aculich @pssachdeva i've deployed this change to the dlab staging hub @ https://dlab-staging.datahub.berkeley.edu

```
(datahub) ➜  datahub git:(dlabrtc_branch) ✗ hubploy deploy dlab hub staging
Activated service account credentials for: [hubploy@ucb-datahub-2018.iam.gserviceaccount.com]
Fetching cluster endpoint and auth data.
kubeconfig entry generated for fall-2019.
Getting updates for unmanaged Helm repositories...
...Successfully got an update from the "https://jupyterhub.github.io/helm-chart/" chart repository
Hang tight while we grab the latest from your chart repositories...
...Successfully got an update from the "grafana" chart repository
Update Complete. ⎈Happy Helming!⎈
Saving 1 charts
Downloading jupyterhub from repo https://jupyterhub.github.io/helm-chart/
Deleting outdated charts
Release "dlab-staging" has been upgraded. Happy Helming!
NAME: dlab-staging
LAST DEPLOYED: Fri Apr 14 18:35:22 2023
NAMESPACE: dlab-staging
STATUS: deployed
REVISION: 915
TEST SUITE: None
```

balajialg committed:
Change the name of env.yml file

balajialg committed:
Add extra configs for groups

balajialg committed:
Making Dlab dir consistent with Stat159

balajialg committed:
Remove managed groups in config

balajialg committed:
Fix hubploy.yaml file

ryanlovett committed:
Quote defaultUrl and move sections up.

defaultUrl is currently not being set in the hub pod's /usr/local/etc/jupyterhub/secrets/.

ryanlovett committed:
Sync additional changes from stat159.

ryanlovett committed:
Set jupyterhub version explicitly.

It was pulling in 1.5.0 without it!

balajialg commented:
@aculich @pssachdeva Thanks to @ryanlovett's support (@shaneknapp's support with hubploys), RTC, and Linux Desktop Environment are enabled in D-Lab staging. Please test the features asap as this change might get overridden when a PR gets merged to prod. Do let us know when you want to merge this to staging (Reminding again - It doesn't have RStudio and R-related packages)

pssachdeva commented:
Thanks @balajialg @shaneknapp @ryanlovett for pushing this forward! 

I am looking at JupyterLab via https://dlab-staging.datahub.berkeley.edu and currently do not see either a "Share" button (for RTC) or a Desktop launcher. I do see a Desktop folder, though:

<img width="1285" alt="Screen Shot 2023-04-26 at 11 45 44 AM" src="https://user-images.githubusercontent.com/9056503/234673636-d9430464-941f-4e3f-8cc9-c5bd64db9568.png">

Is there something I need to toggle to turn these features on? Thanks!

cc @aculich

balajialg commented:
@pssachdeva My best guess is that the changes in this PR which got deployed to D-Lab staging must have been overridden by the latest merges to the prod - https://github.com/berkeley-dsep-infra/datahub/pull/4507. We need to  use hubploy once again to deploy the latest changes to d-lab staging and then you will be able to find the RTC-specific features (which is once again short lived if a new PR gets merged) @shaneknapp @ryanlovett Correct me if I am wrong here.

ryanlovett commented:
@balajialg That's right, the dlab-staging commits were only temporarily deployed. @pssachdeva Until the RTC changes land in dlab/dlab-staging proper (which as I understand it, is waiting for a break in dlab's schedule), you might consider exploring RTC on stat159.datahub.berkeley.edu. That is what dlab's environment is based on.

aculich commented:
Also, this image no longer seems to be tracking the Stat159 regular
<https://stat159.datahub.berkeley.edu/> and Stat159 staging
<https://stat159-staging.datahub.berkeley.edu/> images:
https://stat159-staging.datahub.berkeley.edu/

The desktop is missing and VS code are missing from our image,

[image: image.png]

whereas Stat159 has it:



[image: image.png]

Can you redeploy using their latest staging image that has those components?

Thanks,
 Aaron

On Wed, Apr 26, 2023 at 2:49 PM Ryan Lovett ***@***.***>
wrote:

> @balajialg <https://github.com/balajialg> That's right, the dlab-staging
> commits were only temporarily deployed. @pssachdeva
> <https://github.com/pssachdeva> Until the RTC changes land in
> dlab/dlab-staging proper (which as I understand it, is waiting for a break
> in dlab's schedule), you might consider exploring RTC on
> stat159.datahub.berkeley.edu. That is what dlab's environment is based on.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/berkeley-dsep-infra/datahub/pull/4484#issuecomment-1524087217>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAA5BNKKUNLU62DN2PBTLCLXDGKAFANCNFSM6AAAAAAW65GE3Q>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

ryanlovett commented:
@aculich Our understanding was that we shouldn't move dlab to the stat159 image yet because it doesn't have RStudio, and DLab needs RStudio and the whole R stack for a little while longer for workshops. When would be a good time to make the switch to the new image?

The future dlab image that forked from stat159 can be further developed now. So @pssachdeva can add RStudio and R packages, etc. (via PRs) so that when the changeover happens there is no loss of functionality.
